src/js/ParticleSystem.js

```diff
@@ -0,0 +1,104 @@
+import THREE from 'three';
+
+function applyPointRandomness(r) {
+  let p = new THREE.Vector3(
+    (Math.random() - 0.5),
+    (Math.random() - 0.5),
+    (Math.random() - 0.5))
+    .normalize()
+    .multiplyScalar(r);
+  return p;
+}
+
+class ParticleSystem {
+
+  constructor(scene) {
+    this.emitters = [];
+    this.scene = scene;
+  }
+
+  update(delta) {
+    this.emitters.forEach(emit => {
+      this.updateOne(emit, delta);
+    });
+  }
+
+  createEmitter(options) {
+    let emitter = new THREE.Points();
+
+    emitter.offset = options.offset;
+    emitter.oldPosition = emitter.offset.clone();
+    emitter.bindTo = options.bindTo;
+    emitter.spawnRate = options.spawnRate;
+    emitter.lifetime = options.lifetime;
+    emitter.velocity = options.velocity;
+    emitter.velocityRandomness = options.velocityRandomness;
+    emitter.pointRandomness = options.pointRandomness;
+
+    emitter.geometry = new THREE.Geometry();
+    let toSpawn = Math.ceil(emitter.spawnRate * emitter.lifetime);
+    for (let i = 0; i < toSpawn; i++) {
+      let point = emitter.bindTo.position.clone();
+      let offset = emitter.offset.clone();
+      offset.applyQuaternion(emitter.bindTo.quaternion);
+      if (i === toSpawn - 1) {emitter.oldPosition.copy(point);} // Before randomness
+      offset.add(applyPointRandomness(emitter.pointRandomness));
+      point.add(offset);
+      emitter.geometry.vertices[emitter.iterator] = point;
+      point.velocity = emitter.velocity.clone();
+      point.velocity.multiplyScalar(1 - emitter.velocityRandomness * (2 * Math.random() - 1));
+      emitter.geometry.vertices.push(point);
+    }
+
+    emitter.material = new THREE.PointsMaterial({
+      color: options.color,
+      blending: THREE.AdditiveBlending,
+      transparent: true,
+      map: options.map,
+      size: options.size
+    });
+
+    emitter.iterator = 0;
+    emitter.geometry.computeBoundingSphere();
+
+    this.emitters.push(emitter);
+    this.scene.add(emitter);
+    return emitter;
+  }
+
+  updateOne(emit, delta) {
```

Comment:
UpdateEmitter might be a better name. An emitter could perhaps be it's own class and this could be its update function.